src/wp-includes/comment.php

```diff
@@ -2458,6 +2458,19 @@ function wp_new_comment_notify_postauthor( $comment_id ) {
 	return wp_notify_postauthor( $comment_id );
 }
 
+/**
+ * Send a notification to the post author when a new note is added via the REST API.
+ *
+ * @since 6.9.0
+ *
+ * @param WP_Comment $comment The comment object.
+ */
+function wp_new_comment_via_rest_notify_postauthor( $comment ) {
+	if ( 'note' === $comment->comment_type ) {
+		wp_new_comment_notify_postauthor( (int) $comment->comment_ID );
+	}
+}
```

Comment:
For sage guard, return early if it is not comment object.

```
if ( ! $comment instanceof WP_Comment ) {
	return;
}
```

Reply:
Sure, I will add.